tests/chainer_tests/testing_tests/test_function_link.py

```diff
@@ -118,8 +137,12 @@ def _double_backward_correct(x1, x2, gy1, gy2, ggx1, ggx2):
 # Incoming array types are also checked.
 
 class FuncCorrectlyImplemented(chainer.FunctionNode):
-    def __init__(self, device):
+    def __init__(
+            self, device, allow_grad_outputs_none=False,
+            allow_grad_grad_inputs_none=False):
```

Comment:
Could you elaborate on what you mean with "correctly handled"? If you mean to always allowing `None`, I chose to be explicit, limiting the cases allowing `None` since the callers of the functions are aware of when that happens and we don't want to allow `None` in the other cases.

Reply:
>If you mean to always allowing `None`,

Yes, that's what I mean.

I think I understand the point. Maybe the word "allow" is confusing? (The "correct" function implementation should always allow `None`.)
How about "check" instead?

Reply:
Or perhaps "expect" is better.

Reply:
Fixed to be explicit for each input/output using a tuple of booleans, instead of a single boolean. The parameter is `expect_grad_{outputs,grad_inputs}_none`.